docs/compilers/Visual Basic/Compiler Breaking Changes - post VS2019.md

```diff
@@ -0,0 +1,27 @@
+## This document lists known breaking changes in Roslyn in *Visual Studio 2019 Update 1* and beyond compared to *Visual Studio 2019*.
+
+*Breaks are formatted with a monotonically increasing numbered list to allow them to referenced via shorthand (i.e., "known break #1").
+Each entry should include a short description of the break, followed by either a link to the issue describing the full details of the break or the full details of the break inline.*
+
+1. https://github.com/dotnet/roslyn/issues/38305 
+
+Compiler used to generate incorrect code when a built-in comparison operator producing Boolean? was used
+as an operand of a logical short-circuiting operator used as a Boolean expression.
+For example, for an expression 
+```
+    GetBool3() = True AndAlso GetBool2()
+```
+    and functions
+```
+    Function GetBool2() As Boolean
+        System.Console.WriteLine("GetBool2")
+        Return True
+    End Function
+    Shared Function GetBool3() As Boolean?
+         Return Nothing
+    End Function
+```
+
+it is expected that GetBool2 function going to be called. This is also the expected behavior outside of
```

Comment:
>it is expected that GetBool2 function going to be called [](start = 0, length = 56)

I'm confused why `GetBool2()` should be called. `GetBool3() = True` is a `Boolean` expression that evaluates to `False`.

Reply:
In VB ```GetBool3() = True``` evaluates to Null. Result of comparing unknown (which is what the Null is) with anything (even with another unknown) is unknown, i.e. Null. This is three-valued logic that VB compiler is using, unlike C#.

---
In reply to: [328675106](https://github.com/dotnet/roslyn/pull/38802#discussion_r328675106) [](ancestors = 328675106)